sycl-fusion/passes/kernel-info/SYCLKernelInfo.cpp

```diff
@@ -84,22 +84,30 @@ void SYCLModuleInfoAnalysis::loadModuleInfoFromMetadata(Module &M) {
                     KernelInfo.Args.UsageMask.begin(), getUInt<ArgUsageUT>);
 
     // Operands 3..n: Attributes
-    for (; It != End; ++It) {
-      auto *AIMD = cast<MDNode>(*It);
-      assert(AIMD->getNumOperands() > 1);
+    KernelInfo.Attributes = jit_compiler::SYCLAttributeList{
+        static_cast<size_t>(std::distance(It, End))};
+    std::transform(It, End, KernelInfo.Attributes.begin(), [](const auto &Op) {
+      auto *AIMD = cast<MDNode>(Op);
+      assert(AIMD->getNumOperands() == 4);
       const auto *AttrIt = AIMD->op_begin(), *AttrEnd = AIMD->op_end();
 
       // Operand 0: Attribute name
       auto Name = cast<MDString>(*AttrIt)->getString().str();
+      auto Kind = SYCLKernelAttributeKind::Invalid;
+      if (Name == SYCLKernelAttribute::ReqdWorkGroupSizeName) {
+        Kind = SYCLKernelAttributeKind::ReqdWorkGroupSize;
+      } else if (Name == SYCLKernelAttribute::WorkGroupSizeHintName) {
+        Kind = SYCLKernelAttributeKind::WorkGroupSizeHint;
+      }
+      assert(Kind != SYCLKernelAttributeKind::Invalid);
```

Comment:
```suggestion
      auto Kind = SYCLKernelAttribute::parseKind();
```
Define static member function. Will make sense if the type is moved into the struct